Startup.cs

```diff
@@ -33,16 +33,10 @@ public void ConfigureServices(IServiceCollection services)
                 options.HandleSameSiteCookieCompatibility();
             });
 
-            services.AddOptions();
-            services.AddSignIn(Configuration);
-
-            // Token acquisition service based on MSAL.NET
-            // and chosen token cache implementation
-            services.AddWebAppCallsProtectedWebApi(Configuration, new string[] { "user.read"}).AddInMemoryTokenCaches();
-
-            // // Add APIs
-            // services.AddGraphService(Configuration);
-            // services.AddHttpClient<IArmOperations, ArmApiOperationService>();
+            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+                    .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAd"))
+                        .EnableTokenAcquisitionToCallDownstreamApi(null)
```

Comment:
If you know it already, you might want to pre-specific scopes (optional). This avoids to go though the user challenge afterwards.

Reply:
Yes - the previous sample had on-demand scope, so I left it as is. If you have a strong feeling that we should ask for scopes at login, we can do that also.

Reply:
I think that's fine. If customers complain, we can always add the scope later.